src/librbd/journal/Replay.cc

```diff
@@ -112,6 +112,15 @@ struct C_RefreshIfRequired : public Context {
   C_RefreshIfRequired(I &image_ctx, Context *on_finish)
     : image_ctx(image_ctx), on_finish(on_finish) {
   }
+  virtual ~C_RefreshIfRequired() {
+    delete on_finish;
+  }
+
+  virtual void complete(int r) override {
+    finish(r);
+    on_finish = nullptr;
+    delete this;
+  }
```

Comment:
@dillaman It is just my opinion you can safely ignore. Anyway, I don't like very much when `Context::complete()` is overridden. I consider `Context::complete` method as a part of the interface. And this can easily be avoided here by resetting `on_finish` in `finish()` (and this might look more readable to me).